services/requester/requester.go

```diff
@@ -310,6 +310,11 @@ func (e *EVM) EstimateGas(
 	height uint64,
 	stateOverrides *ethTypes.StateOverride,
 ) (uint64, error) {
+	iterations := 0
+	defer func() {
+		e.collector.GasEstimationIterations(iterations)
+	}()
+
```

Comment:
Maybe might be easier to read if you introduce a closure here:

```go
dryRun := func(gasLimit ??) (??, error) {
   tx.Gas = gasLimit
   result, err := e.dryRunTx(tx, from, height, stateOverrides, nil)
   iterations += 1
   return result, err
}
```

Reply:
That's neat :100: :100: :100: Updated in https://github.com/onflow/flow-evm-gateway/pull/725/commits/c9cb6120d643213948e55a5b71d94322bf6a9b21.